File: Java/media/src/main/java/ar/org/dracul/dto/CreatedByDTO.java
package ar.org.dracul.dto;

public class CreatedByDTO {

    private UserDTO user;
    private String username;

    public UserDTO getUser() {
        return user;
    }

    public void setUser(UserDTO user) {
        this.user = user;
    }

    public String getUsername() {
        return username;
    }

    public void setUsername(String username) {
        this.username = username;
    }

}

File: PHP/DTO/CreatedByDTO.php
<?php

require __DIR__ . "./../DTO/UserDTO.php";

class CreatedByDTO {

    private UserDTO $user;
    private String $username;

    function __construct(stdClass $data) {
        foreach($data as $key => $val) {
            if(property_exists(__CLASS__,$key)) {
                if($key == "user"){
                    $this->user = new UserDTO($val);
                }else{
                    $this->$key = $val;
                }
            }
        }
    }

    public function getUser() {
        return $this->user;
    }

    public function setUser(UserDTO $user) {
        $this->user = $user;
    }

    public function getUsername() {
        return $this->username;
    }

    public function setUsername($username) {
        $this->username = $username;
    }

}
